neurolib/optimize/evolution/evolution.py

```diff
@@ -813,36 +738,59 @@ def loadEvolution(self, fname):
         # the parameter space in the dill
         from neurolib.utils.parameterSpace import ParameterSpace
 
-        pars = ParameterSpace(
-            evolution.parameterSpace.parameterNames,
-            evolution.parameterSpace.parameterValues,
-        )
+        pars = ParameterSpace(evolution.parameterSpace.parameterNames, evolution.parameterSpace.parameterValues,)
 
         evolution.parameterSpace = pars
         evolution.paramInterval = evolution.parameterSpace.named_tuple
-        # # we reinitialize the evolution with the arguments of the saved evolution
-        # evolution.__init__(
-        #     evalFunction=evolution.evalFunction,
-        #     parameterSpace=pars,
-        #     weightList=evolution.weightList,
-        #     model=evolution.model,
-        #     ncores=evolution.ncores,
-        #     POP_INIT_SIZE=evolution.POP_INIT_SIZE,
-        #     POP_SIZE=evolution.POP_SIZE,
-        #     NGEN=evolution.NGEN,
-        #     matingOperator=evolution.matingOperator,
-        #     MATE_P=evolution.MATE_P,
-        #     mutationOperator=evolution.mutationOperator,
-        #     MUTATE_P=evolution.MUTATE_P,
-        #     selectionOperator=evolution.selectionOperator,
-        #     SELECT_P=evolution.SELECT_P,
-        #     parentSelectionOperator=evolution.parentSelectionOperator,
-        #     PARENT_SELECT_P=evolution.PARENT_SELECT_P,
-        #     individualGenerator=evolution.individualGenerator,
-        # )
-
         return evolution
 
+    def _outputToDf(self, pop, df):
+        """Loads outputs dictionary from evolution from the .outputs attribute
+        and writes data into a dataframe.
+
+        :param pop: Population of which to get outputs from.
+        :type pop: list
+        :param df: Dataframe to which outputs are written
+        :type df: pandas.core.frame.DataFrame
+        :return: Dataframe with outputs
+        :rtype: pandas.core.frame.DataFrame
+        """
+        assert len(pop) == len(df), "Dataframe and population do not have same length."
+        # load outputs into dataframe
+        for i, p in enumerate(pop):
+            if hasattr(p, "outputs"):
+                for key, value in p.outputs.items():
+                    # only save floats, ints and arrays
+                    if isinstance(value, (float, int, np.ndarray)):
+                        # save 1-dim arrays
+                        if isinstance(value, np.ndarray):
+                            # to save a numpy array, convert column to object type
+                            if key not in df:
+                                df[key] = None
+                            df[key] = df[key].astype(object)
+                            df.at[i, key] = value
+        return df
```

Comment:
Adding outputs to `evolution.dfEvolution` and `evolution.dfPop` like for `exploration.dfResults`.